orchestrator/main_loops.go

```diff
@@ -261,60 +261,80 @@ func (s *peggyOrchestrator) ValsetRequesterLoop(ctx context.Context) (err error)
 
 func (s *peggyOrchestrator) BatchRequesterLoop(ctx context.Context) (err error) {
 	logger := log.WithField("loop", "BatchRequesterLoop")
+	startTime := time.Now()
+	eightHoursPassed := false
+
 	return loops.RunLoop(ctx, defaultLoopDur, func() error {
 		// get All the denominations
 		// check if threshold is met
 		// broadcast Request batch
 
 		var pg loops.ParanoidGroup
-
 		pg.Go(func() error {
-
 			var unbatchedTokensWithFees []*types.BatchFees
 
 			if err := retry.Do(func() (err error) {
 				unbatchedTokensWithFees, err = s.cosmosQueryClient.UnbatchedTokensWithFees(ctx)
 				return
 			}, retry.Context(ctx), retry.OnRetry(func(n uint, err error) {
 				logger.WithError(err).Errorf("failed to get UnbatchedTokensWithFees, will retry (%d)", n)
-			})); err != nil {
+			}),
+			); err != nil {
 				// non-fatal, just alert
 				logger.Warningln("unable to get UnbatchedTokensWithFees for the token")
 				return nil
 			}
 
-			if len(unbatchedTokensWithFees) > 0 {
-				logger.WithField("unbatchedTokensWithFees", unbatchedTokensWithFees).Debugln("Check if token fees meets set threshold amount and send batch request")
-				for _, unbatchedToken := range unbatchedTokensWithFees {
-					return retry.Do(func() (err error) {
-						// check if the token is present in cosmos denom. if so, send batch request with cosmosDenom
-						tokenAddr := ethcmn.HexToAddress(unbatchedToken.Token)
-
-						var denom string
-						if cosmosDenom, ok := s.erc20ContractMapping[tokenAddr]; ok {
-							// cosmos denom
-							denom = cosmosDenom
-						} else {
-							// peggy denom
-							denom = types.PeggyDenomString(tokenAddr)
-						}
+			if len(unbatchedTokensWithFees) == 0 {
+				logger.Debugln("No outgoing withdraw tx or Unbatched token fee less than threshold")
+				return nil
+			}
 
-						// send batch request only if fee threshold is met.
-						if s.CheckFeeThreshold(tokenAddr, unbatchedToken.TotalFees, s.minBatchFeeUSD) {
-							logger.WithFields(log.Fields{"tokenContract": tokenAddr, "denom": denom}).Infoln("sending batch request")
-							_ = s.peggyBroadcastClient.SendRequestBatch(ctx, denom)
+			logger.WithField("unbatchedTokensWithFees", unbatchedTokensWithFees).Debugln("Check if token fees meets set threshold amount and send batch request")
+			for _, unbatchedToken := range unbatchedTokensWithFees {
```

Comment:
@dbrajovic @kakysha  Maybe I'm not understanding this correctly (even the old code) but isn't it the case currently that only the first `unbatchedToken` would get checked since we `return retry.Do(...` immediately? 

Thus, the first `unbatchedToken` in `unbatchedTokensWithFees` can block successive unbatchedTokens from getting processed if that `unbatchedToken` doesn't meet the fee threshold (even though the successive ones would!). 

This seems like a bug, in both the current and previous implementations. 

Moreover, the logic with `eightHoursPassed` would only apply to the first `unbatchedToken` right? This also would be a bug.

Reply:
Admittedly so, this does look like an already present bug

In other words, this loop makes a batch successfully only **once** and returns early, despite potentially being able to batch others as well. And yes, it definitely seems like it could block batch production on the `peggy` side if the fee threshold isn't met __for every validator__. (which would definitely happen once we go with the minimum 50$ fee hardcoded)